src/listpack_malloc.h

```diff
@@ -39,8 +39,11 @@
 #ifndef LISTPACK_ALLOC_H
 #define LISTPACK_ALLOC_H
 #include "zmalloc.h"
-#define lp_malloc zmalloc
-#define lp_realloc zrealloc
+/* We use zmalloc_usable/zrealloc_usable instead of zmalloc/zrealloc
+ * to ensure the safe invocation of 'zmalloc_usable_size().
+ * See comment in zmalloc_usable_size(). */
+#define lp_malloc zmalloc_usable
+#define lp_realloc zrealloc_usable
```

Comment:
i was thinking of a macro or an inline function that adds `NULL` so that we don't need to change the lines who call this.

another thing i was wondering was maybe to completely eliminate this file (i.e. now that listpack is just a plain part of redis, and not an external library, it can directly use zmalloc).
but then, maybe we won't have a good place to put this comment in, so maybe keep this file, but use the macro i suggested above?